### PR TITLE
feat: Move comments when cut and pasted

### PR DESCRIPTION
**Pull request by marijnh**
Issue https://github.com/dxos/dxos/issues/5040

This implements the feature where cutting and pasting comments will move them in the text. Requires consumers to provide an onMove callback Wires up the comments story to use this, but does not do this for the threads plugin yet.

### CODE REVIEW

File: packages/ui/react-ui-editor/src/extensions/comments.ts [hunk 1]
```diff
@@ -151,6 +155,65 @@ export type CommentsOptions = {
   onHover?: (el: Element, shortcut: string) => void;
 };
 
+const trackPastedComments = (onMove: (threadID: string, cursor: string) => void) => {
```
Comment (richburdon) [inline packages/ui/react-ui-editor/src/extensions/comments.ts:158]: exposing onCut would be useful to allow the app to mark comments that are to be potentially deleted.

File: packages/ui/react-ui-editor/src/extensions/comments.ts [hunk 2]
```diff
@@ -151,6 +155,65 @@ export type CommentsOptions = {
...
+      tracked = {
+        text: view.state.doc.slice(main.from, main.to),
+        comments: inSel.map((range) => ({ from: range.from - main.from, to: range.to - main.from, id: range.id })),
+      };
+    }
+  };
+
+  return [
+    EditorView.domEventHandlers({
+      cut: registerCopy,
```
Comment (richburdon) [inline packages/ui/react-ui-editor/src/extensions/comments.ts:179]: How can we handle undo?
E.g., cut paragraph containing selection, then ctrl-z.
Potentially with history (i.e., multiple times).

**Review by richburdon — APPROVED** [review]

**Comment by richburdon** [comment]
We use https://github.com/marketplace/actions/semantic-pull-request for the PR titles.
I'll change to `feat`.

File: packages/ui/react-ui-editor/src/extensions/comments.ts (hunk 1 above)
Comment (marijnh) [reply packages/ui/react-ui-editor/src/extensions/comments.ts:158]: Aren't those just the comments that cover an empty range?

**Pull request merged**